lib/pavilion/builder.py

```diff
@@ -608,6 +612,16 @@ def _setup_build_dir(self, dest):
                     "Could not copy extra file '{}' to dest '{}': {}"
                     .format(path, dest, err))
 
+    def make_files(self, files, dst):
+        for file in files:
+            # Assumes that each file is a path, e.g., 'somedir/file', or './file'
+            dirname = os.path.dirname(file)
+            Path(dst / dirname).mkdir(parents=True, exist_ok=True)
+            path = Path(file)
+            with open(dst / path, 'w') as f:
+                for line in file:
+                    f.write(line)
```

Comment:
The logic of writing a file to a file object should be its own method.
 1. It will make it easier to write tests of that 'unit'. 
 2. You're going to have to generate these files in two places. Once to hash them under create_build_hash, and once to actually write them once we're creating the build.

The logic of looping over the files can be live outside of it.